spec-tests/src/test/java/org/candlepin/spec/EntitlementCertificateSpecTest.java

```diff
@@ -0,0 +1,634 @@
+        List<CertificateDTO> oldCerts = consumerApi.fetchCertificates(system.getUuid());
+        assertThat(oldCerts).hasSize(1);
+        ConsumerDTO system2 = client.consumers().register(Consumers.random(owner));
+        ApiClient consumerClient = ApiClients.trustedConsumer(system2.getUuid());
+        ConsumerClient consumerApi2 = consumerClient.consumers();
+        assertNotFound(
+            () -> consumerApi2.regenerateEntitlementCertificates(system.getUuid(), null, false));
+    }
+
+    @Test
+    @OnlyInStandalone
+    @DisplayName("should be manually regenerated for a product")
+    public void shouldBeManuallyRegeneratedForAProductStandalone() throws Exception {
+        ProductDTO coolApp = Products.randomEng();
+        coolApp = ownerProductApi.createProductByOwner(owner.getKey(), coolApp);
+        pool = Pools.random(coolApp);
+        pool = ownerApi.createPool(owner.getKey(), pool);
+        consumerApi.bindPool(system.getUuid(), pool.getId(), 1);
+        List<CertificateDTO> oldCerts = consumerApi.fetchCertificates(system.getUuid());
+        assertThat(oldCerts).hasSize(2);
+        AsyncJobStatusDTO result = entitlementsApi.regenerateEntitlementCertificatesForProduct(
+            coolApp.getId(), false);
+        if (result != null) {
+            AsyncJobStatusDTO status = jobApi.waitForJob(result.getId());
+            assertEquals("FINISHED", status.getState());
+        }
+
+        List<CertificateDTO> newCerts = consumerApi.fetchCertificates(system.getUuid());
+        assertEquals(oldCerts.size(), newCerts.size());
+        // System has two certs, but we only regenerated for one product, so the
+        // other serial should have remained the same:
+        assertThat(newCerts)
+            .filteredOn(oldCerts::contains)
+            .hasSize(1);
+    }
+
+    @Test
+    @OnlyInStandalone
+    @DisplayName("should be regenerated when changing existing subscription's end date")
+    public void shouldBeRegeneratedWhenChangingExistingSubsEndDateStandalone() throws Exception {
+        CertificateDTO oldCert = consumerApi.fetchCertificates(system.getUuid()).get(0);
+        pool.setEndDate(pool.getEndDate().plusDays(2L));
+        ownerApi.updatePool(owner.getKey(), pool);
+        CertificateDTO newCert = consumerApi.fetchCertificates(system.getUuid()).get(0);
+        assertNotEquals(oldCert.getSerial().getId(), newCert.getSerial().getId());
+        EntitlementDTO ent = consumerApi.listEntitlements(system.getUuid()).get(0);
+        assertEquals(ent.getEndDate(), pool.getEndDate());
+    }
+
+    @Test
+    @OnlyInStandalone
+    @DisplayName("should delete single entitlement in excess when subscription quantity is decreased")
+    public void shouldDeleteSingleEntInExcessWhenSubQuantityIsDecreasedStandalone() throws Exception {
+        //  this entitlement makes the counts inconclusive
+        consumerApi.unbindAll(system.getUuid());
```

Comment:
This appears in a few tests (I know it's been ported as is from the ruby version), but it's a code smell that can lead to test flakiness. In such a case, where enough of the tests require a piece of data to be created in the beforeEach (in this case a consumer and it's entitlements), while others don't, then we should split one of the two groups in a nested class.

Let's create a nested class only for the tests like this one, that don't want `system` to be pre-entitlement by the `beforeEach`